presto-main/src/main/java/com/facebook/presto/server/PrestoTraceTokenFactory.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import java.util.Optional;
+public class PrestoTraceTokenFactory implements TraceTokenFactory {
+    @Override
+    public Optional<String> getTraceToken(String clientInfo) {
+        // No factory tracing token in open source Presto
+        return Optional.empty();
+    }
+
+    @Override
+    public boolean isTraceTokenFactoryPresent(String clientInfo){
+        // No factory tracing token in open source Presto
+        return false;
+    }
```

Comment:
Do we need two methods ? can we just check if the optional is NOT empty instead of  isTraceTokenFactoryPresent() call ?